blockchain/blockchain.go

```diff
@@ -1763,38 +1756,40 @@ func (bc *BlockChain) CloseBlockSubscriptionLoop() {
 }
 
 // replaceCurrentBlock replaces bc.currentBlock to the given block.
-// func (bc *BlockChain) replaceCurrentBlock(latestBlock *types.Block) {
-// 	bc.mu.Lock()
-// 	defer bc.mu.Unlock()
-//
-// 	if latestBlock == nil {
-// 		logger.Error("no latest block")
-// 		return
-// 	}
-//
-// 	// Return early if it is the first block update.
-// 	currentBlock := bc.CurrentBlock()
-// 	if currentBlock == nil {
-// 		bc.insert(latestBlock)
-// 		logger.Info("inserted an initial block", "blockNumber", latestBlock.NumberU64(),
-// 			"hash", latestBlock.Hash().String(), "stateRoot", latestBlock.Root().String())
-// 		return
-// 	}
-//
-// 	// Don't update current block if the latest block is not newer than current block.
-// 	if currentBlock.NumberU64() >= latestBlock.NumberU64() {
-// 		logger.Debug("")
-// 		return
-// 	}
-//
-// 	// Insert a new block and update metrics
-// 	bc.insert(latestBlock)
-// 	headBlockNumberGauge.Update(latestBlock.Number().Int64())
-// 	bc.stateCache.TrieDB().UpdateMetricNodes()
-//
-// 	logger.Info("inserted a new block", "prevBlockNumber", currentBlock.NumberU64(), "blockNumber",
-// 		latestBlock.NumberU64(), "hash", latestBlock.Hash().String(), "stateRoot", latestBlock.Root().String())
-// }
+func (bc *BlockChain) replaceCurrentBlock(latestBlock *types.Block) {
+	bc.mu.Lock()
+	defer bc.mu.Unlock()
+
+	if latestBlock == nil {
+		logger.Error("no latest block")
+		return
+	}
+
+	// Return early if it is the first block update.
+	currentBlock := bc.CurrentBlock()
+	if currentBlock == nil {
+		bc.insert(latestBlock)
+		logger.Info("inserted an initial block", "blockNumber", latestBlock.NumberU64(),
+			"hash", latestBlock.Hash().String(), "stateRoot", latestBlock.Root().String())
+		return
+	}
+
+	// Don't update current block if the latest block is not newer than current block.
+	if currentBlock.NumberU64() >= latestBlock.NumberU64() {
+		logger.Debug("")
+		return
+	}
+
+	// Insert a new block and update metrics
+	bc.insert(latestBlock)
+	bc.hc.SetCurrentHeader(latestBlock.Header())
+
+	headBlockNumberGauge.Update(latestBlock.Number().Int64())
+	bc.stateCache.TrieDB().UpdateMetricNodes()
+
```

Comment:
```suggestion
	blockTxCountsGauge.Update(int64(latestBlock.Transactions().Len()))
	blockTxCountsCounter.Inc(int64(latestBlock.Transactions().Len()))
```

Reply:
I think the original code is right. :)

Reply:
This one shows the number of processed transactions after the node started, which you can check on the Grafana dashboard.

Reply:
Both metrics are needed to check [1] number of transactions in a block and [2] number of processed transactions since the node started.

Reply:
I though @winnie-byun 's suggestion is replacing the line. I misunderstood. :)

Reply:
@winnie-byun Thanks! I added suggested lines.